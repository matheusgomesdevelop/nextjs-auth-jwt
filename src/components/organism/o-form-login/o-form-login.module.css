.o_form_login {
    display: flex;
    flex-direction: column;
    gap: var(--spacing-20);
}
