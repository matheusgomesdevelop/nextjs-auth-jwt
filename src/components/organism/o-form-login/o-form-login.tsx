import { FormEventHandler } from "react";

// Molecules
import { M_InputWithLabel } from "@/components/molecules";

type errors = { [key: string]: string };

interface I_OFormLogin {
  onSubmit: FormEventHandler;
  errors: errors;
}

export const O_FormLogin = ({ onSubmit, errors }: I_OFormLogin) => {
  return (
    <form onSubmit={onSubmit}>
      <M_InputWithLabel
        labelText="Email"
        type="email"
        name="email"
        placeholder="Informe o seu e-mail"
      />

      {errors["email"] || null}

      <M_InputWithLabel
        labelText="Senha"
        type="password"
        name="password"
        placeholder="Informe a sua senha"
      />

      {errors["password"] || null}
    </form>
  );
};
