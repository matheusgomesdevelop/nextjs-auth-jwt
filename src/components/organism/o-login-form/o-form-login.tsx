import { FormEventHandler } from "react";

// Atoms
import { A_Button } from "@/components/atoms";

// Molecules
import { M_CheckboxWithLabel, M_InputWithLabel } from "@/components/molecules";

import styles from "./o-form-login.module.css";

type errors = { [key: string]: string };

interface I_OFormLogin {
  onSubmit: FormEventHandler;
  errors: errors;
}

export const O_FormLogin = ({ onSubmit, errors }: I_OFormLogin) => {
  return (
    <form onSubmit={onSubmit} className={styles.o_form_login}>
      <M_InputWithLabel
        labelText="Email"
        type="email"
        name="email"
        placeholder="Informe o seu e-mail"
      />

      {errors["email"] || null}

      <M_InputWithLabel
        labelText="Senha"
        type="password"
        name="password"
        placeholder="Informe a sua senha"
      />

      {errors["password"] || null}

      <M_CheckboxWithLabel
        checked={false}
        labelText="Manter-me conectado por 30 dias."
        style={{ margin: "var(--spacing-4) 0" }}
      />

      <A_Button type="submit">Login</A_Button>
    </form>
  );
};
