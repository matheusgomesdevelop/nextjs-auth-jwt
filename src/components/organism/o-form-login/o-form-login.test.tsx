import { FormEventHandler } from "react";

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

// Organism
import { O_FormLogin } from "./o-form-login";

describe("Deve renderizar o o-form-login, corretamente", () => {
  // ======================================================================

  it("Deve renderizar todos campos do formulÃ¡rio", () => {
    // Arrange
    const onSubmit: FormEventHandler = (e) => e.preventDefault();
    const errors = {};

    render(<O_FormLogin onSubmit={onSubmit} errors={errors} />);

    // Act
    const get_m_input_with_label_email = screen.getByTestId(
      "m-input-with-label-email"
    );
    const get_m_input_with_label_password = screen.getByTestId(
      "m-input-with-label-password"
    );

    // Assert
    expect(get_m_input_with_label_email).toBeInTheDocument();
    expect(get_m_input_with_label_password).toBeInTheDocument();
  });

  // ======================================================================
});

// =====================================
