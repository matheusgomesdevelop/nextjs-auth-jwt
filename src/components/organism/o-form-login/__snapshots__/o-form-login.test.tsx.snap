// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o o-form-login, corretamente Deve manter a estrutura visual do componente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="m_input_label"
    data-testid="m-input-with-label-email"
  >
    <label
      className="a_label"
      data-testid="a-label"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="p-inputtext p-component a_input"
      data-pc-name="inputtext"
      data-pc-section="root"
      data-testid="a-input"
      name="email"
      onBeforeInput={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
      placeholder="Informe o seu e-mail"
      type="email"
    />
  </div>
  <div
    className="m_input_label"
    data-testid="m-input-with-label-password"
  >
    <label
      className="a_label"
      data-testid="a-label"
      htmlFor="password"
    >
      Senha
    </label>
    <input
      className="p-inputtext p-component a_input"
      data-pc-name="inputtext"
      data-pc-section="root"
      data-testid="a-input"
      name="password"
      onBeforeInput={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
      placeholder="Informe a sua senha"
      type="password"
    />
  </div>
  <button
    className="p-button p-component a_button"
    data-pc-name="button"
    data-pc-section="root"
    data-testid="a-button"
    disabled={false}
    type="submit"
  >
    Login
  </button>
</form>
`;
